src/pages/VeSDL/index.tsx

```diff
@@ -10,48 +10,228 @@ import {
+        lockedSDLVal
+          .mul(BigNumber.from(leftTimeForUnlock))
+          .div(BigNumber.from(MAXTIME)),
+      )
+    : Zero
+
+  const handleLock = async () => {
+    const unlockTimeStamp = unlockDate ? getUnixTime(unlockDate) : null
+    const hasLockedSDL = !lockedSDLVal.isZero()
+    const shouldCreateLock =
+      !hasLockedSDL && sdlTokenValue.gt(Zero) && unlockTimeStamp
+    const shouldIncreaseAmoutAndLockEnd =
+      hasLockedSDL && sdlTokenValue.gt(Zero) && unlockTimeStamp
+    const shouldIncreaseAmount =
+      hasLockedSDL && sdlTokenValue.gt(Zero) && !unlockTimeStamp
+    const shouldIncreaseLockEnd =
+      hasLockedSDL && sdlTokenValue.isZero() && unlockTimeStamp
+
+    if (!account || !chainId || !votingEscrowContract?.address) return
+    try {
+      if (sdlTokenValue.gt(Zero)) {
+        await checkAndApproveTokenForTrade(
+          sdlContract,
+          votingEscrowContract.address,
+          account,
+          sdlTokenValue,
+          infiniteApproval,
+        )
+      }
+
+      if (shouldCreateLock && unlockTimeStamp) {
+        const txn = await votingEscrowContract?.create_lock(
+          sdlTokenValue,
+          unlockTimeStamp,
+        )
+        void enqueuePromiseToast(chainId, txn.wait(), "createLock")
+      } else if (shouldIncreaseAmoutAndLockEnd) {
+        const txnIncreaseAmount = await votingEscrowContract.increase_amount(
+          sdlTokenValue,
+        )
+        await txnIncreaseAmount.wait()
+        const txnIncreaseUnlockTime =
+          await votingEscrowContract.increase_unlock_time(
+            BigNumber.from(unlockTimeStamp),
+          )
+        await txnIncreaseUnlockTime.wait()
+        enqueueToast("success", "Increased amount and unlock time")
+      } else if (shouldIncreaseAmount) {
+        // Deposit additional SDL into and existing lock
+        const txn = await votingEscrowContract.increase_amount(sdlTokenValue)
+        await txn.wait()
+        void enqueuePromiseToast(chainId, txn.wait(), "increaseLockAmt")
+      } else if (shouldIncreaseLockEnd) {
+        // Extend the unlock time on a lock that already exists
+        const txn = await votingEscrowContract.increase_unlock_time(
+          BigNumber.from(unlockTimeStamp),
+        )
+        await txn.wait()
+        void enqueuePromiseToast(chainId, txn.wait(), "increaseLockEndTime")
+      }
+      void fetchData()
+      resetFormState()
+    } catch (err) {
+      console.log(err)
+    }
   }
 
-  const prevUnlockDate: Date = new Date("2022-5-23")
-  const addLockMos = date ? differenceInMonths(date, prevUnlockDate) : null
+  const handleUnlock = async () => {
+    if (votingEscrowContract && chainId) {
+      const txn = await votingEscrowContract?.force_withdraw()
+      void enqueuePromiseToast(chainId, txn.wait(), "unlock")
+      void fetchData()
+    }
+  }
+
+  const addLockMos =
+    unlockDate && !isNaN(unlockDate.valueOf())
+      ? differenceInMonths(unlockDate, lockEnd || new Date())
```

Comment:
`unlockDate` and `lockEnd` sound like they represent the same thing, what's the difference? Is one a `newUnlockDate` or maybe a `proposedUnlockDate`?